src/keras_format/types.ts

```diff
@@ -44,3 +44,18 @@ export interface PyJsonDict {
  * @see PyJsonDict
  */
 export interface PyJsonArray extends Array<PyJsonValue> {}
+
+/**
+ * A Keras JSON entry representing a Keras object such as a Layer.
+ *
+ * The Keras JSON convention is to provide the `class_name` (e.g., the layer
+ * type) at the top level, and then to place the class-specific configuration in
+ * a `config` subtree.  These class-specific configurations are provided by
+ * subtypes of `PyJsonDict`.  Thus, this `*Serialization` has a type parameter
+ * giving the specific type of the wrapped `PyJsonDict`.
+ */
+export interface BaseSerialization<N extends string, T extends PyJsonDict>
```

Comment:
Pretty sure what I'm asking for is impossible, but wanted to at least mention -- all the types that extend this look an awful lot like the register for serializablility code.  Of course this is spec side, not implementation side.  But would be nice if there were a way to automatically? register classes that extend this.